packages/nuxt/src/app/components/nuxt-island.ts

```diff
@@ -20,7 +20,7 @@ import { remoteComponentIslands, selectiveClient } from '#build/nuxt.config.mjs'
 
 const pKey = '_islandPromises'
 const SSR_UID_RE = /data-island-uid="([^"]*)"/
-const DATA_ISLAND_UID_RE = /data-island-uid/g
+const DATA_ISLAND_UID_RE = /data-island-uid(="[^"]*")?/g
```

Comment:
What about something like:

```suggestion
const DATA_ISLAND_UID_RE = /data-island-uid(?!=)/g
```

Then we wouldn't need to pass a function matcher in...

Reply:
:+1: Just to be sure. Is it possible that vue render something like `data-island-uid=""` instead of `data-island-uid` ?

Reply:
we could update to account for that:

```suggestion
const DATA_ISLAND_UID_RE = /data-island-uid(?!="[^"])/g
```

Reply:
We still need the function matcher in that one 🤔. I guess we could go with your first suggestion. I havn't seen vue rendering `<div attribute="">` instead of simply `<div attribute>` yet

Reply:
nevermind ...
![image](https://github.com/nuxt/nuxt/assets/63512348/6db07edf-b380-4ef2-89f3-fd2d3fda72f8)

Reply:
in that case, this would probably work:

```suggestion
const DATA_ISLAND_UID_RE = /data-island-uid(="")?(?!="[^"])/g
```